src/ws/test-auth.c

```diff
@@ -184,6 +184,67 @@ test_userpass_cookie_check (Test *test,
   g_assert (service != NULL);
   g_assert (response != NULL);
 
+  creds = cockpit_web_service_get_creds (service);
+  g_assert_cmpstr ("me", ==, cockpit_creds_get_user (creds));
+  g_assert_cmpstr ("cockpit", ==, cockpit_creds_get_application (creds));
+  g_assert_null (cockpit_creds_get_password (creds));
+
+  prev_service = service;
+  g_object_unref (service);
+  service = NULL;
+
+  prev_creds = creds;
+  creds = NULL;
+
+  mock_auth_include_cookie_as_if_client (headers, headers, "cockpit");
+
+  service = cockpit_auth_check_cookie (test->auth, "/cockpit", headers);
+  g_assert (prev_service == service);
+
+  creds = cockpit_web_service_get_creds (service);
+  g_assert (prev_creds == creds);
+
+  g_assert_cmpstr ("me", ==, cockpit_creds_get_user (creds));
+  g_assert_null (cockpit_creds_get_password (creds));
+
+  g_hash_table_destroy (headers);
+  g_object_unref (service);
+  json_object_unref (response);
+}
+
+static void
+test_userpass_store_check (Test *test,
```

Comment:
This is almost the same as "test_userpass_cookie_check", let's use a fixture to avoid code duplication.

Reply:
Done